File: 15-min-projects/ale/camel-case/main.cpp
#include <cassert>
#include <string>

std::string dash_to_camel_case(std::string text)
{
    for (int i = 0; i < text.size(); i++) {
        if (text.at(i) == '_' || text.at(i) == '-') {
            text.erase(i, 1);
            try {
                text.at(i) = std::toupper(text.at(i));
            } catch (const std::exception& e) {
                // ignore the _ or - at the end of the sring
            }
        }
    }
    return text;
}

int main()
{
    assert(dash_to_camel_case("this_is_my_variable") == "thisIsMyVariable");
    assert(dash_to_camel_case("join_the_party") == "joinTheParty");
}

File: 15-min-projects/ale/camel-case/main.py
def dash_to_camel_case(text):
    result = ''
    nextUpper = False
    for c in text:
        if c == '_':
            nextUpper = True
        else:
            if nextUpper:
                result += c.upper()
                nextUpper = False
            else:
                result += c
    return result


print(dash_to_camel_case('sum_of_digits'))
